### PR TITLE
handle_auth_response: wrong secret and raise error

### PR DESCRIPTION
**Pull request by bbuccianti**
I'm just a student trying to learn so I really want your feedback guys!
 
Thanks a lot!

### CODE REVIEW

**Comment by dgw** [comment]
Definitely appreciate the initiative! However we already have more comprehensive changes to the database stuff being worked on in #49.

**That said,** if you want to extract your changes to the webhook `handle_auth_response()` function, those look good to have (and aren't covered by any pending PRs, to my knowledge). You could even just modify this PR branch and reword the commit message/PR title—ask for help on IRC if you need it. 😸

File: sopel_modules/github/webhook.py [hunk 1]
```diff
@@ -156,12 +156,14 @@ def handle_auth_response():
     channel = state.split(':')[1]
 
     data = {'client_id': sopel_instance.config.github.client_id,
-             'client_secret': sopel_instance.config.github.secret,
+             'client_secret': sopel_instance.config.github.client_secret,
              'code': code}
```
Comment (dgw) [inline sopel_modules/github/webhook.py]: As long as you're touching this, mind fixing the indent? These continuation lines have one leading space too many. 😁

File: sopel_modules/github/webhook.py [hunk 2]
```diff
@@ -156,12 +156,14 @@ def handle_auth_response():
...
     raw = requests.post('https://github.com/login/oauth/access_token', data=data, headers={'Accept': 'application/json'})
     try:
         res = json.loads(raw.text)
 
+        if 'error' in res:
+            raise ValueError('{err}: {des}'.format(err=res['error'], des=res['error_description'])
```
Comment (dgw) [inline sopel_modules/github/webhook.py]:
```suggestion
            raise ValueError('{err}: {desc}'.format(err=res['error'], desc=res['error_description'])
```

This is just slightly clearer for future maintainers. It took me a moment to figure out what "des" meant. 😅

File: sopel_modules/github/webhook.py (hunk 1 above)
Comment (bbuccianti) [reply sopel_modules/github/webhook.py]: sure!

File: sopel_modules/github/webhook.py (hunk 2 above)
Comment (bbuccianti) [reply sopel_modules/github/webhook.py]: You're right!

**Pull request merged**